src/orbitals.jl

```diff
@@ -1,10 +1,23 @@
-# select the occupied orbitals assuming an insulator
-function select_occupied_orbitals(basis::PlaneWaveBasis, ψ)
-    model  = basis.model
-    n_spin = model.n_spin_components
-    @assert iszero(basis.model.temperature)
-    n_bands = div(model.n_electrons, n_spin * filled_occupation(model), RoundUp)
-    [ψk[:, 1:n_bands] for ψk in ψ]
+# Returns the occupied orbitals and the occupation array without virtual states
+# (or states with small occupation level for metals).
+# threshold is a parameter to distinguish between states we want to keep and the
+# others when using temperature. It is set to 0.0 by default, to treat with insulators.
+function select_occupied_orbitals(basis, ψ, occupation; threshold=0.0)
+    N = [findlast(x -> x > threshold, occk) for occk in occupation]
```

Comment:
A bit nit-picking, but maybe `>=` instead.

Reply:
If we use `>=` then `threshold=0.0" would select all the states for insulators, so I'm not sure that's a good option. However, I agree that `>` is not very good either, but I'm not sure there is a better solution ? Maybe `findlast(x -> x > threshold, occk) - 1` ?

Reply:
Yeah good point. Let's leave it as is.